### PR TITLE
[FEAT/LAYOUT] Create Website Layout

### PR DESCRIPTION
**Pull request by baohuy2k3**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Project

- [x] Website
- [ ] API
- [ ] Documentation
- [ ] Other... (please describe)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other... (please describe)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B
- [x] No tests needed

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Other... (please describe)

### CODE REVIEW

File: apps/website/src/components/Header/navbar.tsx
Comment (foxminchan) [inline apps/website/src/components/Header/navbar.tsx:1]: @baohuy2k3 Adjust the name of the nav item according to the requirements mentioned on zalo

image

> **Warning**
> This code is duplicate

File: apps/website/src/components/Header/index.tsx
Comment (foxminchan) [inline apps/website/src/components/Header/index.tsx:1]: @baohuy2k3 Image must be have attribute lazy loading

Docs: [here](https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading)

File: apps/website/tailwind.config.js
Comment (foxminchan) [inline apps/website/tailwind.config.js:1]: @baohuy2k3 Missing value: 800, 900 & 950 - white-smoke; 950 - dark-moderate-blue

File: apps/website/src/app.tsx
Comment (foxminchan) [inline apps/website/src/app.tsx:1]: @baohuy2k3 this is a layout, not a page

**Review by foxminchan — CHANGES_REQUESTED** [review]
@baohuy2k3 please update the code for each file has comment

File: apps/website/src/components/Header/navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    {name: 'Thanh toán trực tuyến', link: '/', currrent: false , width: 'w-60'},
+    {name: 'Chatbot', link: '/', currrent: false , width: 'w-28'},
+    {name: 'Hỗ trợ', link: '/', currrent: false, subMenu:true, width: 'w-24'},
+  ]
+  const navSupport=[
+    {name: 'Điều khoản sử dụng', link: '/', currrent: false},
+    {name: 'Hướng dẫn sử dụng', link: '/', currrent: false},
+    {name: 'Thông báo', link: '/', currrent: false},
+  ]
+  const showSubMenu = (id:string) => {
```
Comment (qoucname2202) [inline apps/website/src/components/Header/navbar.tsx:14]: @baohuy2k3 Rewrite the section showSubMenu using Hook useState or useToggle

File: apps/website/src/components/Header/navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    const childUl = document.getElementById('subMenu-'+id);
+    if (childUl) {
+      childUl.classList.toggle('hidden');
+    }
+  }
+  const hideSubMenu = (id:string) => {
```
Comment (qoucname2202) [inline apps/website/src/components/Header/navbar.tsx:20]: @baohuy2k3 Rewrite the section hideSubMenu using Hook useState or useToggle

File: apps/website/src/components/Header/navbar.tsx
Comment (qoucname2202) [inline apps/website/src/components/Header/navbar.tsx:1]: @baohuy2k3 Create a folder Navbar and move the navbar.tsx file to that folder

File: apps/website/src/components/Header/index.tsx
Comment (foxminchan) [inline apps/website/src/components/Header/index.tsx:1]: @baohuy2k3
You left extra spaces in the class name

### Example

**Noncompliant:**
" class-name "

**Compliant:**
"class-name"

File: apps/website/src/components/Header/navbar.tsx
Comment (foxminchan) [inline apps/website/src/components/Header/navbar.tsx:1]: @baohuy2k3
You left extra spaces in the class name

### Example

**Noncompliant:**
" class-name "

**Compliant:**
"class-name"

**Review by foxminchan — CHANGES_REQUESTED** [review]
@baohuy2k3 fix code style for each file has comment

File: apps/website/src/components/Header/navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            <FaHome className="  w-5 h-5 text-white" />
+          </Link>
+        </li>
+        {navItems.map((item) => (
+          <li key={item.name} onMouseEnter={() => showSubMenu(item.name)} onMouseLeave={() => hideSubMenu(item.name)} 
+              className={"h-full hover:bg-japonica-500 hover:text-white text-dark-moderate-blue-800 "+item.width}>
+            <Link to={item.link} className="flex items-center justify-center h-full w-full">
+              <span className=" text-xl font-medium ">{item.name}</span>
+            </Link>
+            {item.subMenu && <ul id={"subMenu-"+item.name} className="hidden">
```
Comment (qoucname2202) [inline apps/website/src/components/Header/navbar.tsx:41]: @baohuy2k3 Rewrite the navSupport section into a separate function and call the function

**Review by qoucname2202 — CHANGES_REQUESTED** [review]
@baohuy2k3 Fix code style for each file has comment

File: apps/website/src/app.tsx [hunk 4]
```diff
@@ -1,11 +1,13 @@
 import { Route, Routes } from 'react-router-dom';
 import NxWelcome from './components/NxBase';
 import { NotFound } from './components/NotFound';
+import BasicLayout from './layouts/basic-layout';
 
 export default function App() {
   return (
     <Routes>
       <Route path="/" element={<NxWelcome title="Welcome" />} />
+      <Route path="/home" element={<BasicLayout children={undefined}/>} />
```
Comment (foxminchan) [inline apps/website/src/app.tsx:10]: @baohuy2k3 remove route /home

**Review by foxminchan — CHANGES_REQUESTED** [review]
@baohuy2k3 update code for each file has comment

> **Note**
> Use prettier for format code in [here](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)

**Review by qoucname2202 — APPROVED** [review]
@foxminchan approved

**Review by foxminchan — APPROVED** [review]
Done

**Pull request merged**